Apps/Sandcastle/gallery/Scene Rendering Performance.html

```diff
@@ -0,0 +1,283 @@
+                <input type="text" size="2" data-bind="value: maximumRenderTimeChange, valueUpdate: 'input', enable: timeChangeEnabled">
+            </td>
+        </tr>
+    </table>
+</div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+
+// Create a viewer that won't render a new frame unless
+// updates to the scene require it to reduce overall CPU usage.
+var viewer = new Cesium.Viewer('cesiumContainer', {
+    requestRenderMode : true,
+    maximumRenderTimeChange : Infinity
+});
+
+var scene = viewer.scene;
+scene.debugShowFramesPerSecond = true;
+scene.globe.enableLighting = true;
+viewer.clockViewModel.multiplier = 200;
+
+// Clear scene and set default view.
+var handler;
+function resetScene() {
+    viewer.trackedEntity = undefined;
+    viewer.entities.removeAll();
+    viewer.scene.primitives.removeAll();
+    handler = handler && handler.destroy();
+    scene.skyBox.show = true;
+    scene.camera.flyHome(0.0);
+    scene.requestRender();
+}
+
+// Load a tileset and set the view.
+// No need to call scene.requestRender()
+function loadTilesetScenario() {
+    resetScene();
+
+    var tileset = viewer.scene.primitives.add(new Cesium.Cesium3DTileset({
+        url : 'https://beta.cesium.com/api/assets/1458?access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiIxYmJiNTAxOC1lOTg5LTQzN2EtODg1OC0zMWJjM2IxNGNlYmMiLCJpZCI6NDQsImFzc2V0cyI6WzE0NThdLCJpYXQiOjE0OTkyNjM4MjB9.1WKijRa-ILkmG6utrhDWX6rDgasjD7dZv-G5ZyCmkKg'
+    }));
+
+    tileset.readyPromise.then(function() {
+        var boundingSphere = tileset.boundingSphere;
+        viewer.camera.viewBoundingSphere(boundingSphere, new Cesium.HeadingPitchRange(0.0, -0.5, boundingSphere.radius));
+        viewer.camera.lookAtTransform(Cesium.Matrix4.IDENTITY);
+    }).otherwise(function(error) {
+        throw(error);
+    });
+}
+
+// Load CZML DataSource with a model and set the trackedEntity.
+// No need to call scene.requestRender()
+// Enable and adjust maximum simulation time change to see
+// animations at desired speed.
+function loadCzmlScenario() {
+    resetScene();
+
+    var czml = [{
+        "id" : "document",
+        "name" : "CZML Model",
+        "version" : "1.0"
+    }, {
+        "id" : "aircraft model",
+        "name" : "Cesium Air",
+        "position" : {
+            "cartographicDegrees" : [-77, 37, 10000]
+        },
+        "model": {
+            "gltf" : "../../SampleData/models/CesiumAir/Cesium_Air.glb",
+            "scale" : 2.0,
+            "minimumPixelSize": 128
+        }
+    }];
+
+    var dataSourcePromise = viewer.dataSources.add(Cesium.CzmlDataSource.load(czml));
+
+    dataSourcePromise.then(function(dataSource){
+        viewer.trackedEntity = dataSource.entities.getById('aircraft model');
+    });
+}
+
+// Pick an entity, only rendering when needed.
+function pickingScenario() {
+    resetScene();
+
+    var entity = viewer.entities.add({
```

Comment:
Call `requestRender` after adding this primitive

Reply:
Actually, that might not work since the entity is creating this primitive asynchronously.  How is an end user supposed to handle that case?

Reply:
Since it is being created asynchronously, the return from the web worker will prompt a render automatically. Adding requestRender right after adding it to the scene would be best though, as it will ensure the primitive creation is kicked off.

Reply:
Ah okay, I thought it wasn't added until I moved the camera but I couldn't seem to reproduce it so maybe I just interpreted the timing wrong.

> Adding requestRender right after adding it to the scene would be best though, as it will ensure the primitive creation is kicked off.

I'm confused, will the web worker not start otherwise?

Reply:
I can confirm that the entity does render without changing the camera. The call to primitives.update is still happens during a render since it's not trivial. So the geometry creation wont start until a render happens. This is similar to how we're loading in the globe tiles (see conversation above).